### PR TITLE
1857 add webpack

### PR DESCRIPTION
**Pull request by IagoLast**
#1857 
----

This PR adds webpack to compile the `v4` version of the library.

Due some issues with Webpack that led me to dirty hacks I´ve decided to remove **wax** creating a easy to manage and well known implementation called [carto-grid-json](https://github.com/IagoLast/carto-gridjson) at the moment is in my personal account but as son as @alonsogarciapablo gives the approval we should move it to  a cartodb repo.

This is a very sensitive change, so we should test it well. Please try to break this in any way possible.

[this](https://github.com/CartoDB/cartodb.js/pull/1861/files#diff-4d5e3b7d3b232805aaa6bcccadcf34c5) is the wax subset needed by the `gmaps-cartolayergroup-view` to work that should be removed in the future.

### Checklist
- [x] Google maps compatibility
- [x] Move wax to a external repo and add bundle system to ensure browser compatibility
- [x] Handle grid.json errors cc @rubenmoya 
- [x] Touch events

### CODE REVIEW

**Comment by alonsogarciapablo** [comment]
I'd try to decouple:
- replacing wax (https://github.com/CartoDB/cartodb.js/issues/1735), from
- moving from Grunt + Browserify to Webpack

Couple of questions:

Do we really really need to migrate to Webpack now?
Do we really really have to replace wax by our own implementation now?

**Comment by IagoLast** [comment]
> Do we really really need to migrate to Webpack now?

No

> Do we really really have to replace wax by our own implementation now?

No

**Comment by IagoLast** [comment]
> I'd try to decouple removing wax from the webpack 

This can be done (and is already working in b6eeb715c5985cf2168369032982ac4e4f4220ad but is a big ñapa, exposing globals.

File: src/geo/gmaps/gmaps-cartodb-layer-group-view.js [hunk 1]
```diff
@@ -2,7 +2,9 @@
 var _ = require('underscore');
 var GMapsLayerView = require('./gmaps-layer-view');
 require('leaflet'); // NOTE: Leaflet needs to be required before wax because wax relies on global L internally
-var wax = require('wax.cartodb.js');
+var wax = require('../../../vendor/wax');
+// TODO: Include this from npm
```
Comment (Jesus89) [inline src/geo/gmaps/gmaps-cartodb-layer-group-view.js]: Why is this TODO here?

File: vendor/gridjson.js [hunk 2]
```diff
@@ -0,0 +1,545 @@
...
+
+    return LeafletMap;
+}();
+
+module.exports = LeafletMap;
+
+/***/ })
+/******/ ]);
+});
+//# sourceMappingURL=gridjson.bundle.js.map
```
Comment (ivanmalagon) [inline vendor/gridjson.js]: Is this code in another repo?

This is an already bundled code. Where we can find the original?

**Comment by IagoLast** [comment]
@ivanmalagon @Jesus89 The provisional repo is here https://github.com/IagoLast/carto-gridjson.

If this works OK, we should move the repo to the carto ORG and create a npm module.

**Comment by ivanmalagon** [comment]
When we need to test limit errors.

In your local `Windshaft-cartodb` repo do:
1. Open `layergroup.js` file.
2. To send limit errors when requesting attributes:
    - Go to `LayergroupController.prototype.attributes` method (aprox. L242)
    - Add `err = new Error('canceling statement due to statement timeout');` just before the `if (err) {` statement in internal `finish` function.
3. To send limit errors when requesting grid.json tiles:
    - Go to `LayergroupController.prototype.tileOrLayer` method (aprox. L281)
    - Add this code just before calling `self.finalizeGetTileOrGrid(...)`
```
            if (req.params.format == 'grid.json') {
              err = new Error('canceling statement due to statement timeout');
            }
```

Examples:

For attribute errors
```
LayergroupController.prototype.attributes = function(req, res, next) {
    var self = this;

    req.profiler.start('windshaft.maplayer_attribute');

    step(
        function retrieveFeatureAttributes() {
            var mapConfigProvider = new MapStoreMapConfigProvider(
                self.mapStore, res.locals.user, self.userLimitsApi, res.locals
            );
            self.attributesBackend.getFeatureAttributes(mapConfigProvider, res.locals, false, this);
        },
        function finish(err, tile, stats) {
            req.profiler.add(stats || {});

            err = new Error('canceling statement due to statement timeout');
            if (err) {
                err.label = 'GET ATTRIBUTES';
                next(err);
            } else {
                self.sendResponse(req, res, tile, 200);
            }
        }
    );
};
```

For grid.json errors
```
LayergroupController.prototype.tileOrLayer = function (req, res, next) {
    var self = this;

    step(
        function mapController$getTileOrGrid() {
            self.tileBackend.getTile(
                new MapStoreMapConfigProvider(self.mapStore, res.locals.user, self.userLimitsApi, res.locals),
                req.params, this
            );
        },
        function mapController$finalize(err, tile, headers, stats) {
            req.profiler.add(stats);
            if (req.params.format == 'grid.json') {
              err = new Error('canceling statement due to statement timeout');
            }
            self.finalizeGetTileOrGrid(err, req, res, tile, headers, next);
        }
    );
};
```

**Comment by Jesus89** [comment]
Deployed in ded11.

**Pull request merged**